lib/enum/arrays.rb

```diff
@@ -1,24 +1,37 @@
 class Array
 
+  def initialize(array)
+    @array = array
+  end
+
   def sum_of_numeric_elements
+    @array.select {|num| num.class == Fixnum || num.class == Float}.inject(:+)
   end
 
   def product_of_numeric_elements
+    @array.select {|num| num.class == Fixnum || num.class == Float}.inject(:*)
   end
 
   def even_numeric_elements
+    @array.select {|num| num.class == Fixnum && num.even?}
   end
 
   def odd_numeric_elements
+    @array.select {|num| num.class == Fixnum && num.odd?}
   end
 
   def string_elements
+    @array.select {|word| word.class == String}
   end
 
   def lowercase_string_elements_as_uppercase
+  select { |word| word.is_a?(String) && word.chars.first == word.chars.first.downcase }.map(&:capitalize)
```

Comment:
This method won't work since you're not calling select on anything. You need to have `@array.select{ }` like in the previous methods.